pkg/cli/admin/top/persistentvolumeclaims.go

```diff
@@ -0,0 +1,385 @@
+
+func newOptions(streams genericiooptions.IOStreams) *options {
+	return &options{
+		IOStreams: streams,
+	}
+}
+
+func NewCmdTopPersistentVolumeClaims(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command {
+	o := newOptions(streams)
+	cmd := &cobra.Command{
+		Use:     "persistentvolumeclaims",
+		Aliases: []string{"persistentvolumeclaim", "pvc"},
+		Short:   "Experimental: Show usage statistics for bound persistentvolumeclaims",
+		Long:    topPersistentVolumeClaimsLong,
+		Example: topPersistentVolumeClaimsExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run(cmd.Context(), args))
+		},
+	}
+
+	cmd.Flags().BoolVarP(&o.allNamespaces, "all-namespaces", "A", o.allNamespaces, "If present, list the pvc usage across all namespaces. Namespace in current context is ignored even if specified with --namespace")
+	cmd.Flags().BoolVar(&o.InsecureTLS, "insecure-skip-tls-verify", false, "If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure")
+	cmd.Flags().MarkHidden("insecure-skip-tls-verify")
+	return cmd
+}
+
+func (o *options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	var err error
+	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+	cfg, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	o.ClientConfig = cfg
+
+	routeClient, err := routev1client.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	o.ClientSet, err = kubernetes.NewForConfig(o.ClientConfig)
+	if err != nil {
+		return err
+	}
+	o.getRoute = func(ctx context.Context, namespace string, name string, opts metav1.GetOptions) (*routev1.Route, error) {
+		return routeClient.Routes(namespace).Get(ctx, name, opts)
+	}
+
+	if o.allNamespaces {
+		o.Namespace = metav1.NamespaceAll
+	} else {
+		namespace := cmd.Flag("namespace").Value.String()
+		if len(namespace) != 0 {
+			o.Namespace = namespace
+		}
+	}
+
+	if o.allNamespaces && len(args) != 0 {
+		return fmt.Errorf("a persistentvolumeclaim resource cannot be retrieved by name across all namespaces.")
+	}
+
+	return nil
+}
+
+type persistentVolumeClaimInfo struct {
+	Namespace       string
+	Name            string
+	UsagePercentage string
+}
+
+func (v persistentVolumeClaimInfo) PrintLine(out io.Writer) {
+	printValue(out, v.Namespace)
+	printValue(out, v.Name)
+	printValue(out, v.UsagePercentage)
+}
+
+func (o *options) Run(ctx context.Context, args []string) error {
+	o.BearerToken = o.ClientConfig.BearerToken
+	if len(o.ClientConfig.BearerToken) == 0 {
+		klog.V(4).Info(fmt.Sprintf(`no token is currently in use for this session, attempting to retrieve token from secret "%s" in namespace "%s"`, localhostRecoveryTokenSecret, kubeApiserverNamespace))
+		secret, err := o.ClientSet.CoreV1().Secrets(kubeApiserverNamespace).Get(context.TODO(), localhostRecoveryTokenSecret, metav1.GetOptions{})
```

Comment:
We need to continue with whatever we have defined in kubeconfig and should not try to use any other secret, etc.

Reply:
removed that part